src/platform/macos_iokit/enumeration.rs

```diff
@@ -147,6 +157,73 @@ fn get_children(device: &IoService) -> Result<IoServiceIterator, Error> {
     }
 }
 
+fn get_parent(device: &IoService) -> Result<IoService, Error> {
+    unsafe {
+        let mut handle = 0;
+        let r = IORegistryEntryGetParentEntry(device.get(), kIOServicePlane as *mut _, &mut handle);
+        if r != kIOReturnSuccess {
+            debug!("IORegistryEntryGetParentEntry failed: {r}");
+            return Err(Error::from_raw_os_error(r));
+        }
+
+        Ok(IoService::new(handle))
+    }
+}
+
+fn get_port_number(device: &IoService) -> Option<u32> {
+    get_integer_property::<u32>(device, "PortNum").or_else(|| {
+        if let Ok(parent) = get_parent(device) {
```

Comment:
Is the parent something other than a hub here? How is the line below getting the port for a specific child device when the property is on the parent?

Reply:
This function also references the [libusb code](https://github.com/libusb/libusb/blob/e678b3fad58a508cbd0a6e6dc777fb48346f948b/libusb/os/darwin_usb.c#L1225).